### ISSUE TITLE
Not able to upload annotations on a https end point

### ISSUE THREAD

**Issue by NarenZen** [opencv/cvat#4843]
When I upload annotations zip, I get the below error

```
Mixed Content: The page at 'https://cvat01.myprezent.com/tasks/4' was loaded over HTTPS, but requested an insecure XMLHttpRequest endpoint 'http://cvat01.myprezent.com/api/tasks/4/annotations/1e890edc-d32f-494c-96cc-6935ce828501'. This request has been blocked; the content must be served over HTTPS.
(anonymous) @ cvat-ui.c901a1e9f80749471ab0.min.js:2
cvat-ui.c901a1e9f80749471ab0.min.js:2 Error: tus: failed to upload chunk at offset 0, caused by [object ProgressEvent], originated from request (method: PATCH, url: http://cvat01.myprezent.com/api/tasks/4/annotations/1e890edc-d32f-494c-96cc-6935ce828501, response code: n/a, response text: n/a, request id: n/a).
```

**Comment by bsekachev** [comment]
@NarenZen 

Please, provide a version of CVAT you are using and commit hash.

**Comment by bsekachev** [comment]
I will close the issue, please do not hesitate to reopen if you can provide the requested information

**Comment by NarenZen** [comment]
Latest code., which is there on github

**Comment by loidy** [comment]
I have the same problem

**Comment by gumshoes** [comment]
Same issue with 2.2.0, unfortunately this makes CVAT difficult to host behind HTTPS.
For now I have resorted to hosting HTTPS for users that don't do any uploads and a special entry point over HTTP on a VPN for users that have to do uploads.
Anyone have a better solution?

**Comment by bsekachev** [comment]
On ``https://app.cvat.ai`` we have https protocol and everything works fine there, so, need to understand where specifics are.

**Comment by azhavoro** [comment]
@gumshoes If you have deployed CVAT behind a reverse proxy, make sure your proxy is correctly forwarding the HTTP_X_FORWARDED_PROTO header, this header is required for CVAT to determine the correct protocol.

**Comment by loidy** [comment]
@azhavoro we are using nginx reverse proxy with `proxy_set_header X-Forwarded-Proto "https";` set and it still doesn't work

**Comment by azhavoro** [comment]
@loidy Ok, please add `--log.level=DEBUG` to other command arguments here https://github.com/opencv/cvat/blob/develop/docker-compose.yml#L162, restart with `docker compose up -d` and try to upload file. Atfter dump logs with `docker logs traefik > traefik.log` and attach here.

**Comment by gumshoes** [comment]
My setup is CVAT deployed via the Helm in git running in AWS EKS cluster with ingress provided by Traefik and an AWS NLB that does the SSL termination.
The browser error is:
`Mixed content: The Page at 'https://<REDACTED>/projects?page=1' was loaded over HTTPS, but requested an insecure XMLHttpRequest endpoint 'http://<REDACTED>/api/projects/backup/<UUID>'.  This request has been blocked; the content must be served over HTTPS.`

**Comment by jakiro2017** [comment]
Hello  I found out the reason was traefik rewrite the header, if not trust the  reverse proxy in front of it.

TLDR: 
add "- --entryPoints.web.forwardedHeaders.trustedIPs=ip_proxy/mask" to traefik in docker-compose.yml

Ref:
https://github.com/traefik/traefik/issues/5551

**Comment by jevansbio** [comment]
Upgrading to 2.5 has led to us having similar issues. The patch request that is made when adding a task is changed from https to http. If it is changed by traefik, I am uncertain how to chek. The request never even seems to hit traefik. I don't know how smokescreen interacts with all this either.

**Comment by jevansbio** [comment]
I have this issue when trying to upload files as well with cvat running on https.

**Comment by azhavoro** [comment]
I believe that the cause and solution have been found, the issue can be closed
@jevansbio try this https://github.com/opencv/cvat/issues/4843#issuecomment-1414760683

**Comment by jevansbio** [comment]
I had tried it, but it didn't work as in my case the request never actually hit traefik, the browser seemed to change it and block before it got anywhere. I fixed it by adding:

<meta http-equiv="Content-Security-Policy" content="upgrade-insecure-requests" />

To the header of the template.

**Comment by baudneo** [comment]
Hi, I am experiencing the same. I will try both methods and report back.

I use CloudFlare and CloudFlare tunnels to proxy to cvat.

Thanks @azhavoro for pointing me towards a fix!

**Comment by baudneo** [comment]
Adding CloudFlare ips to the traefik command worked for me.

`- '--entryPoints.web.forwardedHeaders.trustedIPs=173.245.48.0/20,103.21.244.0/22,103.22.200.0/22,103.31.4.0/22,141.101.64.0/18,108.162.192.0/18,190.93.240.0/20,188.114.96.0/20,197.234.240.0/22,198.41.128.0/17,162.158.0.0/15,104.16.0.0/13,104.24.0.0/14,172.64.0.0/13,131.0.72.0/22'`
